ArmPkg/Drivers/CpuDxe/CpuMmuCommon.c

```diff
@@ -146,14 +147,60 @@ SetGcdMemorySpaceAttributes (
       RegionLength = MemorySpaceMap[Index].BaseAddress + MemorySpaceMap[Index].Length - RegionStart;
     }
 
+    // MU_CHANGE START: Sync AARCH64 GCD Capabilities with Page Table
+    // Always add RO, RP, and XP, as all memory is able to support these types and they are critical

```

Comment:
I can edit this comment for clarity. I mean that some of the GCD capabilities are capabilities of the underlying HW (and so that we cannot set it globally on all memory), but that `EFI_MEMORY_[RO|RP|XP]` are software constructs and all memory is technically capable of supporting these attributes (although we may not want to actually set the attributes).

This is the model that the x86 CpuDxe driver has taken, we support all of these software capabilities on any memory range and then update the attributes as appropriate.

Reply:
A minor edit might be "capable of supporting" instead of "able to support".